### DISCUSSION TITLE
kitty command for logging a window's output in real time

### DISCUSSION THREAD

**Discussion by resolritter** [kovidgoyal/kitty#6326] [Q&A] [answered]
The [`script` utility from util-linux](https://man7.org/linux/man-pages/man1/script.1.html) can be used to log a terminal session's output to a given file. I am using it as follows:

> script --flush --log-out /tmp/log.txt

I guess it would possible to implement an equivalent `kitty` command, as in:

> kitty @ record --match id:$WINDOW_ID --output /tmp/log.txt

I wanted to figure out how to implement such command by myself, thus I began reading `kitty`'s source code. Still, I was unable to figure out an implementation path. Any guidance is appreciated.

Note: I am aware of `kitty @ get-text`, but it doesn't work the same way as `script` does. I want to have a program which **continuously** and **immediately** flushes any output to a given file, i.e. start the program once and have it log until stopped.

**Comment by kovidgoyal** [comment, marked as answer]
kitty --dump-bytes and note this will log only output not input unlike
script. And you should only use it with a single kitty window at a time.

**Reply by resolritter** [reply]
I was not aware that `--dump-bytes` existed. Thank you for letting me know.

Based on what I understood from the source code, apparently it's not possible to filter the bytes for a specific window, therefore it's not proper for my use-case.

https://github.com/kovidgoyal/kitty/blob/af65b68ac3eec07bc81d8cdfb046cc2bc10998c5/kitty/boss.py#L237

That being said, I suppose that implementing such filtering feature is outside of the scope of this discussion. I already got the answer I was looking for, so I consider this to be solved. Thanks again.
